### DISCUSSION TITLE
is it possible in a post to select the body from the value of another parameter?

### DISCUSSION THREAD

**Discussion by DSO01** [fastapi/fastapi#10940] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class Item(BaseModel):
    name: str
    description: str | None = None
    cpu: int
    ram: int

@deploy.post('/create')
def deploy_item_in_vra(item:Item,catalog,env:VraEnv = VraEnv.dev):
    response = vra_deploy_item(env=env,item=item)
    return response
```

### Description

is it possible in fastapi to select a body according to the value of another parameter? 

### Operating System

Windows

### FastAPI Version

0.105.0

### Pydantic Version

2.5.2

### Python Version

3.10.6

### Additional Context

   is it possible in fastapi to select a body according to the value of another parameter?

**Comment by heysaeid** [comment]
I don't understand why such an action is necessary, but you should have it in the following form (if I have understood the meaning correctly):

```python
from config import settings

class Item1(BaseModel):
    name: str
    description: str | None = None
    cpu: int
    ram: int
    env: str

class Item2(BaseModel):
    name: str
    description: str | None = None
    env: str
 
models = {
    "dev": Item1,
    "stage": Item2,
}

@app.post("/create")
def deploy_item_in_vra(item: models[settings.env] = Depends()):
    response = vra_deploy_item(item=item)
    return response
```

**Reply by DSO01** [reply]
Hello heysaeid, thankyou for your answer.
 I am using fastapi as a catalog of tools and not as an API. that is the reason why I would like to have functions in the swagger parameters. 

I try to use your code but I have an issue with the package config I can not find "settings"  module 
the line "from config import settings" failed even if I install the config package "pip install config"

**Reply by heysaeid** [reply]
In general, by adding the setting in the following manner, you can make use of it. However, for more information, please refer to https://fastapi.tiangolo.com/advanced/settings/


```python
from pydantic import BaseSettings

class Settings(BaseSettings):
    env: str = "DEV"

settings = Settings()
```

**Reply by DSO01** [reply]
thankyou, unfortunatly this can not be a solution for me cause settings variable is initialized each time the main:app is reinitiated
what I need is a variable that is reinitiated each time I launch a specific function
I have found an other way to solve it. when using main:app --reload 
my first function creates a file that initiate the pydantic Models needed in the second function 
the file when created or modified activates the reload of the API  and the second function can accès the new model as a parameter type

the issue now is for my prod catalog which is started without reload in a windows service.
is it possible to reload a pydantic model from a file  in a child function of the API?

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
